addons/mail/static/src/core/common/message_model.js

```diff
@@ -169,7 +172,7 @@ export class Message extends Record {
     }
 
     get editable() {
-        if (!this.allowsEdition) {
+        if (this.isEmpty || !this.allowsEdition) {
```

Comment:
If we're about removing arbitrary code, why don't we allow editing back "deleted" message to normal message while we are at it?

This would allow removing last occurrence of `nonEmptyMessages` too.

Reply:
I wish to remove `isEmpty` too but that might be more difficult.

Reply:
A lot of `isEmpty` can be removed though, in `firstUnreadMessage` for example, in `orderedMessages`, in message_patch (in web; I don't even know why we had that condition), etc

Then we can see if we can remove/adapt the few remaining occurrences.

Reply:
> If we're about removing arbitrary code, why don't we allow editing back "deleted" message to normal message while we are at it?

Editing a message is to fix mistakes but this has limitation e.g. it doesn't notify new people of edited message even with newly included mentions. This is not desirable (can be obnoxious and create very challenging UI/UX), when we can just post a new message.

It's not intuitive to have a message being deleted and revive it for editing. This put too much power in editing message, which goes against making new things in conversation by posting new messages. Keep in mind edition / deletion must stay exceptional at the appreciation of the author.

Reply:
> A lot of `isEmpty` can be removed though, in `firstUnreadMessage` for example, in `orderedMessages`, in message_patch (in web; I don't even know why we had that condition), etc

I removed `isEmpty` in `firstUnreadMessage` and `orderedMessages`. For the `web/message_patch.xml`, the overridden template sucks and duplicate a bit the usage of `message.body`. In theory this is empty but I put the condition to avoid showing `subtype_description`, which in theory such a message could be deleted by user thus we must not show `subtype_description`.